File: Day-11/VP2.py
class VP2:
    def validPalindrome(self, s: str) -> bool:
        left,rgt=0,len(s)-1
        while(left<rgt):
            if s[left]==s[rgt]:
                left+=1
                rgt-=1
            else:
                return self.isvalid(s,left+1,rgt) or self.isvalid(s,left,rgt-1)
        return True
    
    def isvalid(self,s,left,rgt):
        while(left<rgt):
            if s[left]!=s[rgt]:
                return False
            left+=1
            rgt-=1
        return True


        

File: Day-11/VP2.java
public class VP2 {
 
    public boolean validPalindrome(String s) {
        int lft=0,rgt=s.length()-1;
        while(lft<rgt)
        {
            if(s.charAt(lft)==s.charAt(rgt))
            {
                lft++;
                rgt--;
            }
            else{
                return isvalid(s,lft+1,rgt) || isvalid(s,lft,rgt-1);
            }
        }
        return true;
    }
    public boolean isvalid(String s,int lft,int rgt)
    {
        while(lft<rgt)
        {
            if(s.charAt(lft)!=s.charAt(rgt))
            {
                return false;
            }
            lft++;
                rgt--;
        }
        return true;
    }
}

